ui/appui-layout-react/src/appui-layout-react/widget-panels/PanelSections.tsx

```diff
@@ -0,0 +1,155 @@
+
+  const getPercentage = React.useCallback((min: number, max: number, current: number) => {
+    const range = max - min;
+    const adjusted = Math.max(min, Math.min(max, current));
+    if (adjusted === min)
+      return 0;
+    if (adjusted === max)
+      return 100;
+    const percent = ((adjusted - min) * 100) / (range);
+    return percent;
+  }, []);
+
+  const updatePanelSize = React.useCallback(
+    (event: PointerEvent) => {
+      if (!containerRef.current)
+        return;
+
+      const parentPanel = containerRef.current.closest(".nz-widgetPanels-panel");
+      const sectionToResize = containerRef.current.parentElement as HTMLElement;
+      if (parentPanel && sectionToResize) {
+        const rect = parentPanel.getBoundingClientRect();
+        const percent = getPercentage(
+          isHorizontal ? rect.left : rect.top,
+          isHorizontal ? rect.right : rect.bottom,
+          isHorizontal ? event.clientX : event.clientY,
+        );
+
+        dispatch({
+          type: "PANEL_SET_SPLITTER_VALUE",
+          side,
+          percent,
+        });
+      }
+    }, [getPercentage, isHorizontal, side, dispatch]);
+
+  const handlePointerMove = React.useCallback((event: Event): void => {
+    if (splitterProcessingActiveRef.current) {
+      event.preventDefault();
+      event.stopPropagation();
+      updatePanelSize(event as PointerEvent);
+    }
+  }, [updatePanelSize]);
+
+  const handlePointerUp = React.useCallback((event: Event) => {
+    updatePanelSize(event as PointerEvent);
+    event.preventDefault();
+    event.stopPropagation();
+    containerRef.current?.ownerDocument.removeEventListener("pointermove", handlePointerMove);
+    containerRef.current?.ownerDocument.removeEventListener("pointerup", handlePointerUp);
+  }, [handlePointerMove, updatePanelSize]);
+
+  const handlePointerDownOnSplitter = React.useCallback(
+    (event: React.PointerEvent) => {
+      if (containerRef.current) {
+        containerRef.current?.ownerDocument.addEventListener("pointermove", handlePointerMove);
+        containerRef.current?.ownerDocument.addEventListener("pointerup", handlePointerUp);
+        splitterProcessingActiveRef.current = true;
+        event.preventDefault();
+        event.stopPropagation();
+      }
+    }, [handlePointerMove, handlePointerUp]);
+
+  const className = isHorizontal ? "nz-horizontal-panel-splitter" : "nz-vertical-panel-splitter";
+  return (
+    <div ref={containerRef} className={className} onPointerDown={handlePointerDownOnSplitter} />
+  );
+}
+
+/** @internal */
+export function PanelSections() {
+  const side = React.useContext(PanelSideContext);
+  assert(!!side);
+
+  const panel = useLayout((state) => {
+    const p = state.panels[side];
+    const span = isHorizontalPanelState(p) ? p.span : false;
+    return {
+      collapsed: p.collapsed,
+      splitterPercent: p.splitterPercent,
+      widgets: p.widgets,
+      resizable: p.resizable,
+      span,
+    };
```

Comment:
While a copy of the other version, most member are not used in this context.
```suggestion
    const span = isHorizontalPanelState(p) ? p.span : false;
    return {
      splitterPercent: p.splitterPercent,
      widgets: p.widgets,
    };
```